scenario/bookmark.go

```diff
@@ -0,0 +1,103 @@
+package scenario
```

Comment:
I'm not entirely sure if this package is the best location of this file. Maybe we should consider adding another package for this and similar files in the future.

Reply:
But I like the breaking out of this functionality this way!

Reply:
was considering senseobjects, but was afraid of circular dependencies in this case.

Reply:
but agree on general sentiment, migh want to create different subpackage to scenario to separate what is actually actions from helpers and common sections

Reply:
maybe something like 
scenario/helpers
scenario/common
scenario/actions?

Reply:
should probably be a separate refactor from this though

Reply:
Something like that, I don't know exactly. But there are some other pieces that could go there, such as CanAddToCollection and ResolveDestinationSpace

Reply:
yes agree